nomt/src/beatree/mod.rs

```diff
@@ -245,6 +247,46 @@ impl Tree {
     }
 }
 
+/// A change in the value associated with a key.
+#[derive(Clone)]
+pub enum ValueChange {
+    /// The key-value pair is deleted.
+    Delete,
+    /// A new value small enough to fit in a leaf is inserted.
+    Insert(Vec<u8>),
+    /// A new value which requires an overflow page is inserted.
+    InsertOverflow(Vec<u8>, ValueHash),
+}
+
+impl ValueChange {
+    /// Create a [`ValueChange`] from an option, determining whether to use the normal or overflow
+    /// variant based on size.
+    pub fn from_option<T: crate::ValueHasher>(maybe_value: Option<Vec<u8>>) -> Self {
+        match maybe_value {
+            None => ValueChange::Delete,
+            Some(v) => Self::insert::<T>(v),
+        }
+    }
+
+    /// Create an insertion, determining whether to use the normal or overflow variant based on size.
+    pub fn insert<T: crate::ValueHasher>(v: Vec<u8>) -> Self {
+        if v.len() > MAX_LEAF_VALUE_SIZE {
```

Comment:
Tbf, not sure if this is a good place for such a decision.

Reply:
I wasn't sure either, but I would also prefer not to leak `MAX_LEAF_VALUE_SIZE` anywhere else in the codebase. It should be part of the `beatree` module, and `ValueChange` is used everywhere within `beatree/mod.rs`. This feels like the least-bad option. Do you have another suggestion?